Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm

```diff
@@ -1145,10 +1142,18 @@ static FloatSize fullscreenPreferencesScreenSize(CGFloat preferredWidth)
         m_validationBubble->show();
 }
 
-void WebPageProxy::hardwareKeyboardAvailabilityChanged(bool keyboardIsAttached)
+void WebPageProxy::hardwareKeyboardAvailabilityChanged()
 {
+    auto hardwareKeyboardState = currentHardwareKeyboardState();
+    if (hardwareKeyboardState == internals().hardwareKeyboardState)
+        return;
+
+    internals().hardwareKeyboardState = hardwareKeyboardState;
+
+    protectedPageClient()->hardwareKeyboardAvailabilityChanged();
+
     updateCurrentModifierState();
-    m_process->send(Messages::WebPage::HardwareKeyboardAvailabilityChanged(keyboardIsAttached), webPageID());
+    m_process->send(Messages::WebPage::HardwareKeyboardAvailabilityChanged(currentHardwareKeyboardState()), webPageID());
```

Comment:
Why can't we reuse the `hardwareKeyboardState` local here?

Reply:
Will make both changes, as soon as the active `api-ios` EWS run is finished.